addons/product/models/product_pricelist.py

```diff
@@ -109,7 +109,10 @@ def _compute_price_rule(self, products_qty_partner, date=False, uom_id=False):
         """
         self.ensure_one()
         if not date:
-            date = self._context.get('date') or fields.Date.today()
+            context_date = self._context.get('date')
+            if context_date:
+                context_date = fields.Date.from_string(context_date)
+            date = context_date or fields.Date.today()
         date = fields.Date.to_date(date)  # boundary conditions differ if we have a datetime
```

Comment:
After the `fields.Date.to_date` call, `date` is always a `datetime.date` anyway (even if `date` was a string), I don't see why the earlier conversion is needed.

Reply:
@Feyensv This one has to be done on v11.